File: Lab 01 - Vetores e inteiros/lab01.py
def banner_set(u, b):
    for i in range(len(u)):
        b_in_u = False
        for j in range(len(b)):
            if u[i] == b[j]:
                b_in_u = True
                break
        if not b_in_u:
            u[i] = 0
    return u


def mobius_multiplication(u, M):
    for i in range(len(u)):
        if i % 2 == 1:
            u[i] *= M
    return u


def cyclic_movement(u, P):
    aux = []
    for i in range(P):
        aux.append(u[len(u)-P+i])
    for i in range(len(u)-1, P-1, -1):
        u[i] = u[i-P]
    for i in range(P):
        u[i] = aux[i]
    return u


def vector_inversion(u):
    # Inversão manual:
    # for i in range(len(u)//2):
    #     aux = u[i]
    #     u[i] = u[len(u)-1-i]
    #     u[len(u)-1-i] = aux
    # return u

    # Inversão automática:
    u.reverse()
    return u


def operate(u):
    operation = [int(i) for i in input().split()]
    if operation[0] == 1:
        return banner_set(u, operation[2:])
    if operation[0] == 2:
        return mobius_multiplication(u, operation[1])
    if operation[0] == 3:
        return cyclic_movement(u, operation[1])
    if operation[0] == 4:
        return vector_inversion(u)


def print_vector(u):
    for i in range(len(u)):
        if i != len(u) - 1:
            print(u[i], end=' ')
        else:
            print(u[i])


def main():
    T = int(input())
    u = [int(i) for i in input().split()]
    operation_count = int(input())
    for _ in range(operation_count):
        u = operate(u)
        print_vector(u)


if __name__ == "__main__":
    main()

File: Lab 01 - Vetores e inteiros/lab01.c
#include <stdio.h>

/**
 * \brief     Remove de um vetor inteiros não presentes em outro vetor.
 * \param[in] u: Vetor de inteiros n1, n2, ..., nT.
 * \param[in] T: Tamanho do vetor u.
 */
void banner_set(int u[], int T) {
    int B;
    scanf("%d", &B);
    int b[B];
    for (int i = 0; i < B; i++)
        scanf("%d", &b[i]);
    for (int i = 0; i < T; i++) {
        int b_in_u = 0;
        for (int j = 0; j < B; j++) {
            if (u[i] == b[j]) {
                b_in_u = 1;
                break;
            }
        }
        if (!b_in_u)
            u[i] = 0;
    }
}

/**
 * \brief     Multiplica por um int os valores em índices ímpares de um vetor.
 * \param[in] u: Vetor de inteiros n1, n2, ..., nT.
 * \param[in] T: Tamanho do vetor u.
 */
void mobius_multiplication(int u[], int T) {
    int M;
    scanf("%d", &M);
    for (int i = 0; i < T; i++) {
        if (i % 2 == 1)
            u[i] *= M;
    }
}

/**
 * \brief     Desloca cada elemento de um vetor em P índices para a direita.
 * \param[in] u: Vetor de inteiros n1, n2, ..., nT.
 * \param[in] T: Tamanho do vetor u.
 */
void cyclic_movement(int u[], int T) {
    int P;
    scanf("%d", &P);
    int aux[P];
    for (int i = 0; i < P; i++)
        aux[i] = u[T-P+i];
    for (int i = T - 1; i >= P; i--)
        u[i] = u[i-P];
    for (int i = 0; i < P; i++)
        u[i] = aux[i];
}

/**
 * \brief     Inverte um vetor.
 * \param[in] u: Vetor de inteiros n1, n2, ..., nT.
 * \param[in] T: Tamanho do vetor u.
 */
void vector_inversion(int u[], int T) {
    int aux;
    for (int i = 0; i < T / 2; i++) {
        aux = u[i];
        u[i] = u[T-1-i];
        u[T-1-i] = aux;
    }
}

/**
 * \brief     Realiza uma operação sob um vetor.
 * \param[in] u: Vetor de inteiros n1, n2, ..., nT.
 * \param[in] T: Tamanho do vetor u.
 */
void operate(int u[], int T) {
    int operation;
    scanf("%d", &operation);
    switch (operation) {
    case 1:
        banner_set(u, T);
        break;
    case 2:
        mobius_multiplication(u, T);
        break;
    case 3:
        cyclic_movement(u, T);
        break;
    case 4:
        vector_inversion(u, T);
        break;
    }
}

/**
 * \brief     Imprime um vetor.
 * \param[in] u: Vetor de inteiros n1, n2, ..., nT.
 * \param[in] T: Tamanho do vetor u.
 */
void print_vector(int u[], int T) {
    for (int i = 0; i < T; i++)
        if (i != T-1)
            printf("%d ", u[i]);
        else
            printf("%d\n", u[i]);
}

/**
 * \brief Executa o programa principal, que realiza operações um vetor.
 */
int main(void) {
    int T;
    scanf("%d", &T);
    int u[T];
    for (int i = 0; i < T; i++)
        scanf("%d", &u[i]);
    int operation_count;
    scanf("%d", &operation_count);
    for (int i = 0; i < operation_count; i++) {
        operate(u, T);
        print_vector(u, T);
    }
    return 0;
}
